packages/hadron-document/src/element.js

```diff
@@ -486,13 +498,42 @@ class Element extends EventEmitter {
     return (this.key === ID && !this.isAdded()) || !this.isRemovable();
   }
 
+  /**
+   * Determine if the value has been decrypted via CSFLE.
+   *
+   * @returns {Boolean} If the value was encrypted on the server and is now decrypted.
+   */
+  isValueDecrypted() {
+    return this.decrypted;
+  }
+
+  /**
+   * Detemine if this value or any of its children were marked
+   * as having been decrypted with CSFLE.
+   *
+   * Warning: This does *not* apply to the children of these elements!
```

Comment:
> Maybe I'm misunderstanding this function, but aren't we evaluating here if the children of this element have decrypted children? 

I’ve made the comments a bit more explicit – it returns whether a child of this element has been decrypted. That would not apply to a child of a decrypted field, because decrypted fields generally don’t have decrypted children.

I will admit that I’ve never actually checked what happens when nested encryption happens on nested fields. I don’t think anybody would ever really do this on purpose, but I’ve opened COMPASS-5705 to remind myself to look into that, and how we would handle it in Compass. :slightly_smiling_face: 

> Should this warning be for `isValueDecrypted`?

It also applies there, that’s true :slightly_smiling_face: Added it.